### ISSUE TITLE
Consumes a lot of memory during large body size requests

### ISSUE THREAD

**Issue by juri-t** [newrelic/go-agent#795]
[NOTE]: # ( ^^ Provide a general summary of the issue in the title above. ^^ )

## Description

When we send a request with a large body size, it consumes a lot of memory because NewRelic is putting all the bodies in memory with io.ReadAll. Depending on the server's memory, this can bring down the server with an Out of Memory error.

https://github.com/newrelic/go-agent/blob/master/v3/newrelic/transaction.go#L255

[NOTE]: # ( Describe the problem you're encountering. )
[TIP]:  # ( Do NOT give us access or passwords to your New Relic account or API keys! )

## Steps to Reproduce

```golang
package main

import (
	"github.com/labstack/echo/v4"
	"github.com/newrelic/go-agent/v3/integrations/nrecho-v4"
	"github.com/newrelic/go-agent/v3/newrelic"
	"io"
)

func main() {
	e := echo.New()

	app, _ := newrelic.NewApplication(
		newrelic.ConfigAppName("App Name"),
		newrelic.ConfigLicense("Licence Key"),
		func(cfg *newrelic.Config) {
			cfg.Enabled = true
		},
	)

	e.POST("/", func(ec echo.Context) error {
		io.Copy(io.Discard, ec.Request().Body)
		return ec.NoContent(200)
	}, nrecho.Middleware(app))
	e.Start(":3000")
}
```

POST requests for large files created by mkfile, etc., consume a lot of memory.

[NOTE]: # ( Please be as specific as possible. )

## Expected Behavior
Does not consume large amounts of memory.

[NOTE]: # ( Tell us what you expected to happen. )

## NR Diag results

[NOTE]: # ( Provide any other relevant log data. )

## Your Environment

[TIP]:  # ( Include as many relevant details about your environment as possible including the running version of New Relic software and any relevant configurations. )

## Reproduction case

[TIP]: # ( Link a sample application that demonstrates the issue. )

## Additional context

[TIP]:  # ( Add any other context about the problem here. )

**Comment by mirackara** [comment]
Hello @juri-t, thanks for the report! We are currently investigating this.

**Comment by iamemilio** [comment]
This is released in the 3.28.0 version of the go agent!
